tests/queries/0_stateless/01881_join_on_conditions_merge.sql.j2

```diff
@@ -72,8 +72,8 @@ SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t1.key; -- { serverErro
 SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t2.key; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT, INVALID_JOIN_ON_EXPRESSION }
 SELECT * FROM t1 JOIN t2_nullable as t2 ON t2.key == t2.key2 AND (t1.id == t2.id OR isNull(t2.key2)); -- { serverError INVALID_JOIN_ON_EXPRESSION }
 SELECT * FROM t1 JOIN t2 ON t2.key == t2.key2 OR t1.id == t2.id; -- { serverError INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 JOIN t2 ON (t2.key == t2.key2 AND (t1.key == t1.key2 AND t1.key != 'XXX' OR t1.id == t2.id)) AND t1.id == t2.id; -- { serverError INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 JOIN t2 ON t2.key == t2.key2 AND t1.key == t1.key2 AND t1.key != 'XXX' AND t1.id == t2.id OR t2.key == t2.key2 AND t1.id == t2.id AND t1.id == t2.id; -- { serverError NOT_IMPLEMENTED }
+SELECT * FROM t1 JOIN t2 ON (t2.key == t2.key2 AND (t1.key == t1.key2 AND t1.key != 'XXX' OR t1.id == t2.id)) AND t1.id == t2.id; -- { serverError NOT_IMPLEMENTED }
+SELECT * FROM t1 JOIN t2 ON t2.key == t2.key2 AND t1.key == t1.key2 AND t1.key != 'XXX' AND t1.id == t2.id OR t2.key == t2.key2 AND t1.key == t2.key2 AND t1.key == t2.key; -- { serverError NOT_IMPLEMENTED }
```

Comment:
This is optimized to ` ALL INNER JOIN default.t2 AS __table2 ON (__table2.key = __table2.key2) AND (__table1.id = __table2.id)` due to the second part of the original query. The optimized join expression is an easy one, even partial_merge can execute it. Thus I had to change the query.